examples/scripts/counter.js

```diff
@@ -126,23 +128,32 @@ WoT.produce({
     .then((thing) => {
     console.log("Produced " + thing.getThingDescription().title);
     // init property values
-    thing.writeProperty("count", 0);
-    thing.writeProperty("lastChange", (new Date()).toISOString());
+    count = 0;
+    lastChange = (new Date()).toISOString();
+    // set property handlers
+    thing.setPropertyReadHandler("count", () => {
+        return new Promise((resolve, reject) => {
+            resolve(count);
+        });
+    });
+    thing.setPropertyReadHandler("lastChange", () => {
+        return new Promise((resolve, reject) => {
+            resolve(lastChange);
+        });
+    });
```

Comment:
Using async it can be written like this:
```js
thing.setPropertyReadHandler("lastChange", async () => lastChange });
```

Reply:
My personal preference would be staying with promise return but I do not have a strong opinion. Either way is fine.

To get yet another opinion....  @egekorkan what do you think?

Reply:
I also prefer the version @relu91 wrote above :)

Reply:
Agreed and updated accordingly